### ISSUE TITLE
CTE (Common Table Expressions) within subqueries

### ISSUE THREAD

**Issue by maoxingda** [reata/sqllineage#484]
**Hope to support CTE (Common Table Expressions) in subqueries. Thank you.**

```sql
insert into
    public.tgt_tbl1
(
    id
)
with
    t1 as
    (
        with
            t2 as
            (
                select
                    id
                from
                    public.src_tbl1
            )
        select
            id
        from
            t2
    )
select
    id
from
    t1
;
```

**Comment by maoxingda** [comment]
https://github.com/reata/sqllineage/issues/476

**Comment by reata** [comment]
This potentially will be solved when we solve #476 and #481. Mark it as duplicate and keep it open for now.

**Comment by reata** [comment]
#476 (CTE within subquery) is now fixed. But looks like here we have CTE within CTE and they're not handled in a unified way. We still need to fix this issue separately.
